source/grid/tria.cc

```diff
@@ -9810,19 +9866,34 @@ Triangulation<dim, spacedim>::create_triangulation(
         create_triangulation(v, cells, subcelldata, *this);
 
       this->update_reference_cell_types();
-
-#ifndef DEAL_II_WITH_SIMPLEX_SUPPORT
-      Assert(this->all_reference_cell_types_are_hyper_cube(),
-             ExcMessage(
-               "A cell with invalid number of vertices has been provided."));
-#endif
     }
   catch (...)
     {
       clear_despite_subscriptions();
       throw;
     }
 
+  if (this->all_reference_cell_types_are_hyper_cube())
```

Comment:
I have rebased. Now `all_reference_cell_types_are_hyper_cube()` is actually in use :)